### ISSUE TITLE
Bug: for-direction rule doesn't detect wrong direction when update clause uses SequenceExpression (comma operator)

### ISSUE THREAD

**Issue by Kuldeep2822k** [eslint/eslint#20726]
### Environment

Node version: v22.x
npm version: 10.x
Local ESLint version: latest (main)
Global ESLint version: N/A
Operating System: Windows 11

### What parser are you using?

Default (Espree)

### What did you do?

<summary>Configuration</summary>

```js
/* eslint for-direction: "error" */
// Case 1: counter moves in wrong direction inside a SequenceExpression
for (let i = 10; i < 20; i--, j++) {}

// Case 2: should NOT report — net effect on counter is +1 (valid loop)
for (let i = 10; i < 20; i--, i += 2) {}

// Case 3: counter modified twice, opposite directions — net effect is 0
for (let i = 10; i < 20; i--, i++) {}
```

### What did you expect to happen?

1. Case 1 → error: The update clause in this loop moves the variable in the wrong direction.
2. Case 2 → No error (net effect on i is +1, loop terminates normally)
3. Case 3 → No error / direction unknown (counter modified twice with opposing directions; net effect is statically ambiguous)

### What actually happened?

No output for any of the three cases. The rule passes silently without reporting an error.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IGZvci1kaXJlY3Rpb246IFwiZXJyb3JcIiAqL1xuXG5mb3IgKGxldCBpID0gMTA7IGkgPCAyMDsgaS0tLCBqKyspIHt9Iiwib3B0aW9ucyI6eyJydWxlcyI6eyJmb3ItZGlyZWN0aW9uIjpbImVycm9yIl19LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

The tricky part is handling cases where the counter appears more than once in the sequence:

for(i = 10; i < 20; i--, i++) — net effect is 0 → infinite loop, but statically ambiguous
for(i = 10; i < 20; i--, i += 2) — net effect is +1 → valid loop, must NOT report
Proposed safe behavior: If the counter appears in more than one expression within the sequence, treat direction as unknown and skip. Only report when exactly one expression in the sequence modifies the counter in the wrong direction.

I am willing to submit a pull request for this issue (PR #20701 has a working implementation pending team review of the edge cases above).

**Comment by mdjermanovic** [comment]
> // Case 1: counter moves in wrong direction inside a SequenceExpression
> for (let i = 10; i < 20; i--, j++) {}

I think this should be reported, and I think it's fine to treat this as a semver-minor bug fix (false negative).

@eslint/eslint-team thoughts?

> The tricky part is handling cases where the counter appears more than once in the sequence:
> 
> for(i = 10; i < 20; i--, i++) — net effect is 0 → infinite loop, but statically ambiguous for(i = 10; i < 20; i--, i += 2) — net effect is +1 → valid loop, must NOT report Proposed safe behavior: If the counter appears in more than one expression within the sequence, treat direction as unknown and skip. Only report when exactly one expression in the sequence modifies the counter in the wrong direction.

The proposed safe behavior sounds good to me.

**Comment by JoshuaKGoldberg** [comment]
Agreed all around. 👍

**Comment by SwetaTanwar** [comment]
Makes sense to me as well. Marking as accepted.
